components/core/src/clp_s/Utils.cpp

```diff
@@ -48,16 +59,106 @@ bool FileUtils::find_all_files(std::string const& path, std::vector<std::string>
     return true;
 }
 
-bool FileUtils::validate_path(std::vector<std::string> const& paths) {
-    bool all_paths_exist = true;
-    for (auto const& path : paths) {
-        if (false == boost::filesystem::exists(path)) {
-            SPDLOG_ERROR("'{}' does not exist.", path.c_str());
-            all_paths_exist = false;
+namespace {
+/**
+ * Determines if a directory is a multi-file archive.
+ * @param path
+ * @return true if this directory is a multi-file archive, false otherwise
+ */
+bool directory_is_multi_file_archive(std::string_view const path) {
```

Comment:
Might be worthwhile, yeah. Could we leave it to the follow-up PR though? My next PR applies a small diff to this function that I might forget otherwise.